pkg/apis/eksctl.io/v1alpha5/vpc.go

```diff
@@ -97,6 +97,13 @@ type (
 		ExtraCIDRs []*ipnet.IPNet `json:"extraCIDRs,omitempty"`
 		// for pre-defined shared node SG
 		SharedNodeSecurityGroup string `json:"sharedNodeSecurityGroup,omitempty"`
+		// Automatically add security group rules to and from the default
+		// cluster security group and the shared node security group.
+		// This allows unmanagd nodes to communicate with the control plane
+		// and managed nodes.
+		// Defaults to `true`
+		// +optional
+		ManageSharedNodeSecurityGroupRules *bool `json:"manageSharedNodeSecurityGroupRules,omitempty"`
```

Comment:
My preference would be to name this `DisableSharedNodeSecurityGroupRules` (or something else that signifies this disables a default feature) which is `false` by default, and only allow setting it if `VPC.SharedNodeSecurityGroup` is provided.

Reply:
Yeah, that make sense! Would `DisableManagedSharedSecurityGroup Rules` be any better?

If we're saying that this should only matter if you're explicitly providing the shared node security group, then maybe it should actually remain as is and have a default of `false`? That would be more backwards compatible.

I realize I'm arguing against myself from earlier, but if we're breaking symmetry with the implicitly created group, we could keep things backwards compatible (we would have to test for both conditions anyway).

I still think it probably makes more sense to default to manage by default, but I'm a sample size of 1.

Reply:
>I realize I'm arguing against myself from earlier, but if we're breaking symmetry with the implicitly created group, we could keep things backwards compatible (we would have to test for both conditions anyway).


I kind of agree with your previous point about managing SG rules by default (when a custom SG is provided) at the expense of breaking backwards compatibility. If we're going this route, we should also log a message saying that the default behaviour has changed and SG rules will be added, and that it can be disabled by setting `disableManagedSharedSecurityGroup: true`

Reply:
Any recommendation on where such a warning should live?

Reply:
I think it can go in the cluster task creation.